app/src/main/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt

```diff
@@ -35,22 +79,56 @@ class SearchLocationActivity : AppCompatActivity() {
                 // Do your search
                 searchLocationViewModel.saveSearch(query)
                 val startMainActivity = Intent(this@SearchLocationActivity, MainActivity::class.java).setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
-
                 if (fromSearchFragment) {
                     val searchBundle = Bundle()
                     searchBundle.putBoolean(TO_SEARCH, true)
                     startMainActivity.putExtras(searchBundle)
                 }
-
                 startActivity(startMainActivity)
-
                 return false
             }
 
             override fun onQueryTextChange(newText: String): Boolean {
                 return false
             }
         })
+
+        current_location.setOnClickListener {
+            progress_bar.visibility = View.VISIBLE
+            val permission = ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION)
+            if (permission != PackageManager.PERMISSION_GRANTED) {
+                ActivityCompat.requestPermissions(this,
+                    arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION), MY_PERMISSIONS_REQUEST)
+            }
+            LocationServices.getFusedLocationProviderClient(applicationContext).requestLocationUpdates(locationRequest, locationCallback, null)
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        when (requestCode) {
+            MY_PERMISSIONS_REQUEST -> {
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    LocationServices.getFusedLocationProviderClient(applicationContext).requestLocationUpdates(locationRequest, locationCallback, null)
+                }
+            }
+        }
+    }
+
+    fun LocationName(latitude: Double, longitude: Double): String {
+        try {
+            val geocoder = Geocoder(applicationContext, Locale.getDefault())
+            val addresses: List<Address>
+            addresses = geocoder.getFromLocation(latitude, longitude, 10)
+            for (address: Address in addresses) {
+                if (address.locality != null && address.locality.length > 0) {
+                    return address.locality
+                }
+            }
+        } catch (e: IOException) {
+            Log.d("SearchLocationActivity", "Error fetching name")
```

Comment:
Use Timber, I think we are using it. Also, this is not a debug log. This is an exception